WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift

```diff
@@ -138,16 +145,16 @@ final class HubMenuViewModel: ObservableObject {
     }
 
     private func setupPOSElement() {
-        let isBetaFeatureEnabled = generalAppSettings.betaFeatureEnabled(.pointOfSale)
-        let eligibilityChecker = POSEligibilityChecker(cardPresentPaymentsOnboarding: CardPresentPaymentsOnboardingUseCase(),
-                                                       siteSettings: ServiceLocator.selectedSiteSettings.siteSettings,
-                                                       currencySettings: ServiceLocator.currencySettings,
-                                                       featureFlagService: featureFlagService)
-        if isBetaFeatureEnabled && eligibilityChecker.isEligible() {
-            posElement = PointOfSaleEntryPoint()
-        } else {
-            posElement = nil
-        }
+        cardPresentPaymentsOnboarding.refreshIfNecessary()
+        Publishers.CombineLatest(generalAppSettings.betaFeatureEnabledPublisher(.pointOfSale), posEligibilityChecker.isEligible)
+            .map { isBetaFeatureEnabled, isEligibleForPOS in
+                if isBetaFeatureEnabled && isEligibleForPOS {
+                    return PointOfSaleEntryPoint()
+                } else {
+                    return nil
+                }
+            }
```

Comment:
nit: it might be clearer to use a filter then a map:

```suggestion
            .filter { isBetaFeatureEnabled, isEligibleForPOS in
                isBetaFeatureEnabled && isEligibleForPOS
            }
            .map { _ in
                PointOfSaleEntryPoint()
            }
```

Reply:
If the downstream value is `nil` from `isBetaFeatureEnabled && isEligibleForPOS  = false` => `PointOfSaleEntryPoint ` from `isBetaFeatureEnabled && isEligibleForPOS  = true`, then when `isBetaFeatureEnabled && isEligibleForPOS  = false`, would filter + map set the value back to `nil`? I'll do a quick experiment and update.

Reply:
Tested in [Playground](https://gist.github.com/jaclync/3b77c8d0fdfa6481721558320b72fe38) today, if we apply the filter, the view model doesn't get reset back to `nil` when the entry point is not eligible anymore.